tests/countries/test_thailand.py

```diff
@@ -117,72 +115,9 @@ def test_special_holidays(self):
-            ("2022-01-01", "วันขึ้นปีใหม่"),
-            ("2022-01-03", "ชดเชยวันขึ้นปีใหม่"),
-            ("2022-01-08", "วันเด็กแห่งชาติ"),
-            ("2022-01-13", "วันการบินแห่งชาติ"),
-            ("2022-01-14", "วันอนุรักษ์ทรัพยากรป่าไม้ของชาติ"),
-            ("2022-01-16", "วันครู"),
-            ("2022-01-17", "วันพ่อขุนรามคำแหงมหาราช"),
-            ("2022-01-18", "วันกองทัพไทย"),
-            ("2022-02-03", "วันทหารผ่านศึก"),
-            ("2022-02-16", "วันมาฆบูชา"),
-            ("2022-02-26", "วันศิลปินแห่งชาติ"),
-            ("2022-03-08", "วันสตรีสากล"),
-            ("2022-04-06", "วันจักรี"),
-            ("2022-04-13", "วันสงกรานต์"),
-            ("2022-04-14", "วันสงกรานต์"),
-            ("2022-04-15", "วันสงกรานต์"),
-            ("2022-05-01", "วันแรงงานแห่งชาติ"),
-            ("2022-05-02", "ชดเชยวันแรงงานแห่งชาติ"),
-            ("2022-05-04", "วันฉัตรมงคล"),
-            ("2022-05-13", "วันพืชมงคล"),
-            ("2022-05-15", "วันวิสาขบูชา"),
-            ("2022-05-16", "ชดเชยวันวิสาขบูชา"),
-            (
-                "2022-06-03",
-                "วันเฉลิมพระชนมพรรษาสมเด็จพระนางเจ้าสุทิดา พัชรสุธาพิมลลักษณ พระบรมราชินี",
-            ),
-            ("2022-07-13", "วันอาสาฬหบูชา"),
-            ("2022-07-14", "วันเข้าพรรษา"),
-            ("2022-07-15", "วันหยุดพิเศษ (เพิ่มเติม)"),
-            (
-                "2022-07-28",
-                (
-                    "วันเฉลิมพระชนมพรรษาพระบาทสมเด็จพระปรเมนทรรามาธิบดี"
-                    "ศรีสินทรมหาวชิราลงกรณ พระวชิรเกล้าเจ้าอยู่หัว"
-                ),
-            ),
-            ("2022-07-29", "วันหยุดพิเศษ (เพิ่มเติม)"),
-            ("2022-08-12", "วันเฉลิมพระชนมพรรษาสมเด็จพระบรมราชชนนีพันปีหลวง; วันแม่แห่งชาติ"),
-            ("2022-08-18", "วันวิทยาศาสตร์แห่งชาติ"),
-            ("2022-09-28", "วันพระราชทานธงชาติไทย"),
-            (
-                "2022-10-13",
-                "วันคล้ายวันสวรรคตพระบาทสมเด็จพระบรมชนกาธิเบศร มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร",
-            ),
-            ("2022-10-14", "วันหยุดพิเศษ (เพิ่มเติม)"),
-            ("2022-10-23", "วันปิยมหาราช"),
-            ("2022-10-24", "ชดเชยวันปิยมหาราช"),
-            ("2022-11-08", "วันลอยกระทง"),
-            (
-                "2022-12-05",
-                (
-                    "วันคล้ายวันเฉลิมพระชนมพรรษาพระบาทสมเด็จพระบรมชนกาธิเบศร "
-                    "มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร; วันชาติ; วันพ่อแห่งชาติ"
-                ),
-            ),
-            ("2022-12-10", "วันรัฐธรรมนูญ"),
-            ("2022-12-12", "ชดเชยวันรัฐธรรมนูญ"),
-            ("2022-12-30", "วันหยุดพิเศษ (เพิ่มเติม)"),
-            ("2022-12-31", "วันสิ้นปี"),
-        )
-
     def test_new_years_day(self):
-        self.assertHoliday(f"{year}-01-01" for year in range(1941, 2058))
+        name = "วันขึ้นปีใหม่"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1941, 2058)))
```

Comment:
I'm thinking of doing this for Laos and Cambodia later as well, since they're also affected by this at the moment - I'll create another PRs for them later.